tests/quickinspectortest.cpp

```diff
@@ -258,18 +203,123 @@ private slots:
+        QVERIFY(updatedSpy.isValid());
+
+//        {
+//            connect(remoteView, &RemoteViewInterface::requestUpdate, this, [&]() {
+//                qWarning("--> Frame requested: %i", requestedSpy.count());
+//            });
+
+//            connect(remoteView, &RemoteViewInterface::frameUpdated, remoteView, [&]() {
+//                qWarning("<-- Frame updated: %i", updatedSpy.count());
+//            });
+//        }
+
+        // Testing static scene only send one frame
+        for (int i = 0; i < 5; ++i) {
+//            qWarning("*** Starting static test: %i", i);
+            remoteView->setViewActive(true);
+            // Activating the view trigger an update request
+            QVERIFY(waitForSignal(&updatedSpy, true));
+            QCOMPARE(requestedSpy.count(), 1);
+            QCOMPARE(updatedSpy.count(), 1);
+
+            requestedSpy.clear();
+            updatedSpy.clear();
+
+            view()->update();
+            QVERIFY(waitForSignal(&updatedSpy, true));
+            for (int j = 0; j < 5; ++j) {
+                remoteView->clientViewUpdated();
+                QTest::qWait(throttlerInterval);
+            }
+            QTest::qWait(throttlerInterval * 2);
+            QCOMPARE(requestedSpy.count(), 1);
+            QCOMPARE(updatedSpy.count(), 1);
+
+            requestedSpy.clear();
+            updatedSpy.clear();
+
+            remoteView->setViewActive(false);
+//            qWarning("*** Ending static test: %i", i);
+        }
+
+        // Our animation properties
+        const qreal animationInterval = throttlerInterval;
+        const qreal animationDuration = 1400.0;
+        // Qml try to render @ ~60fps
+        const qreal maxPossibleQmlRequests = animationDuration / 1000.0 * 60.0;
+        const qreal maxPossibleThrottledRequests = qMin(maxPossibleQmlRequests, animationDuration / throttlerInterval);
+
+        for (int i = 0; i < 5; i++) {
+//            qWarning("*** Starting dynamic test: %i", i);
+            remoteView->setViewActive(true);
+            // Activating the view trigger an update request
+            QVERIFY(waitForSignal(&updatedSpy, true));
+            QCOMPARE(requestedSpy.count(), 1);
+            QCOMPARE(updatedSpy.count(), 1);
+
+            requestedSpy.clear();
+            updatedSpy.clear();
+
+            // We rotate our rendering every animationInterval-ms for animationDuration-ms
+            rootItem->setProperty("interval", qRound(animationInterval));
+            rootItem->setProperty("animated", true);
+            QTest::qWait(qRound(animationDuration));
+
+            // Wait to process pending requests...
+            rootItem->setProperty("animated", false);
+            QTest::qWait(qRound(animationDuration));
+
+            QVERIFY(requestedSpy.count() <= qRound(maxPossibleThrottledRequests * 1.05));
+            QVERIFY(updatedSpy.count() <= qRound(maxPossibleThrottledRequests * 1.05));
```

Comment:
We never call clientViewUpdated() here, so shouldn't this be always 1 here, no matter how much the source actually changed? That would actually be a valid test-case, independent of what I suspect this code tried to test :)